engine/src/main/java/org/hibernate/validator/internal/util/annotation/ConstraintAnnotationDescriptor.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.util.annotation;
+
+import java.lang.annotation.Annotation;
+import java.util.Map;
+
+import javax.validation.ConstraintTarget;
+import javax.validation.Payload;
+
+import org.hibernate.validator.internal.metadata.core.ConstraintHelper;
+
+/**
+ * @author Marko Bekhta
+ */
+public class ConstraintAnnotationDescriptor<A extends Annotation> extends AnnotationDescriptor<A> {
+
+	public ConstraintAnnotationDescriptor(A annotation) {
+		super( annotation );
+	}
+
+	public ConstraintAnnotationDescriptor(AnnotationDescriptor<A> descriptor) {
+		super( descriptor );
+	}
+
+	public String message() {
```

Comment:
Second thoughts on this one. As `ConstraintAnnotationDescriptor` is a JavaBean, let's stick to the bean convention and use `getMessage()`. Same for the others below.